### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: 'reduce'

### ISSUE THREAD

**Issue by CLG68** [mseok/PIGNet2#4]
Hi,

There seems to be a dependency missing as I get:

FileNotFoundError: [Errno 2] No such file or directory: 'reduce'
when I run one of your example:

(pignet2) christian@christian-linux02:/media/christian/VS1/VS/Scoring_PIGNet2$ python src/exe/predict.py ./src/ckpt/pda_0.pt -p examples/protein.pdb -l examples/ligand1.sdf -o examples/case2.txt

**Comment by CLG68** [comment]
I used:
conda install kalininalab::reduce
to install reduce 4.7.

now I get this:

RuntimeError: nvrtc: error: invalid value for --gpu-architecture (-arch)

nvrtc compilation failed: 

#define NAN __int_as_float(0x7fffffff)
#define POS_INFINITY __int_as_float(0x7f800000)
#define NEG_INFINITY __int_as_float(0xff800000)


template<typename T>
__device__ T maximum(T a, T b) {
  return isnan(a) ? a : (a > b ? a : b);
}

template<typename T>
__device__ T minimum(T a, T b) {
  return isnan(a) ? a : (a < b ? a : b);
}

extern "C" __global__
void fused_sub_exp(float* tsrc_1, float* tsrc_max_9, float* aten_exp) {
{
  if (512 * blockIdx.x + threadIdx.x<22544 ? 1 : 0) {
    float v = __ldg(tsrc_1 + 512 * blockIdx.x + threadIdx.x);
    float v_1 = __ldg(tsrc_max_9 + 512 * blockIdx.x + threadIdx.x);
    aten_exp[512 * blockIdx.x + threadIdx.x] = expf(v - v_1);
  }
}
}

**Comment by CLG68** [comment]
My GPU and cuda is well detected:

(pignet2) christian@christian-linux02:/media/christian/VS1/VS/Scoring_PIGNet2$ nvidia-smi
Wed Aug 14 17:55:19 2024       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.90.07              Driver Version: 550.90.07      CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4060 Ti     Off |   00000000:07:00.0  On |                  N/A |
|  0%   44C    P8             15W /  165W |     461MiB /  16380MiB |      1%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
So I'm not sure what is the problem

**Comment by sangyeon-hits** [comment]
Hello, CLG68.

As the message says, you need an external executable [reduce](https://github.com/rlabduke/reduce) for protonating proteins in a PDB file.

Please refer to the hyperlink to install the program.

You can find the relevant guideline in the [README for preprocessing](../blob/main/dataset/preprocess)

**Comment by CLG68** [comment]
Thank you,

I used conda install kalininalab::reduce; it installed the latest version (reduce 4.7).  Once that was solved, I got a problem related to my GPU, as cuda11.1 does not support ada lovelace GPU.  So I installed cuda toolkit 11.8 and the updated versions of torch and related tools that fit cuda11.8.  Still, I have problems with torch-sparse.  I also tried with the versions suggested in the requirements.txt file but on a PC with a RTX3070, which should be ok with cuda11.1; I still have problems with torch-sparse.  

I'll try to make a minimal environment with only torch-related files and to solve this problem and then install the remaining files.  I'll let you know if it works.

Best,
Christian
